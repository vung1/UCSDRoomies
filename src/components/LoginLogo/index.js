import React from "react";
import { Text, Image, View, StyleSheet } from "react-native";

const appLogoPNG = require("../../../assets/images/appLogo.png");

function LoginLogo() {
  return (
    <View style={styles.logoContainer}>
      <Image source={appLogoPNG} style={styles.image} />
      <Text style={{ fontSize: 24, fontFamily: "AmericanTypewriter-Bold" }}>
        {" "}
        UCSD{" "}
      </Text>
      <Text style={{ fontSize: 24, fontFamily: "AmericanTypewriter-Bold" }}>
        {" "}
        Roomies{" "}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  logoContainer: {
    width: 200,
    height: 200,
    backgroundColor: "white",
    alignItems: "center",
    borderRadius: 100,
  },
  image: {
    width: 90,
    height: 90,
    marginTop: "8%",
    marginBottom: "3%",
  },
});

export default LoginLogo;
