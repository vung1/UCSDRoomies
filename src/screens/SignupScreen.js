import React from "react";
import { View, Text, TextInput, StyleSheet } from "react-native";
import { LinearGradient } from "expo-linear-gradient";

import LoginLogo from "../components/LoginLogo";
import SignupButton from "../components/SignupButton";
import BackArrow from "../components/BackArrow";

function SignupScreen({ navigation }) {
  const [state, setState] = React.useState({
    email: "",
    password: "",
    repassword: "",
  });

  const [emialInputBorder, setEmailInputBorder] = React.useState(0);
  const [passwordInputBorder, setPasswordInputBorder] = React.useState(0);
  const [repasswordInputBorder, setRepasswordInputBorder] = React.useState(0);

  const [emailErrorMsg, setEmailMsg] = React.useState("");
  const [passwordErrorMsg, setPasswordMsg] = React.useState("");
  const [repasswordErrorMsg, setRePasswordMsg] = React.useState("");

  const showMessages = (emailMsg, passwordMsg, repasswordMsg) => {
    setEmailMsg(emailMsg);
    setPasswordMsg(passwordMsg);
    setRePasswordMsg(repasswordMsg);
    if (emailMsg === "") {
      setEmailInputBorder(0);
    } else {
      setEmailInputBorder(2);
    }

    if (passwordMsg === "") {
      setPasswordInputBorder(0);
    } else {
      setPasswordInputBorder(2);
    }

    if (repasswordMsg === "") {
      setPasswordInputBorder(0);
    } else {
      setPasswordInputBorder(2);
    }
  };

  return (
    <LinearGradient colors={["#74AED6", "#247DCF"]} style={styles.background}>
      {/* Back arrow to login screen */}
      <View style={styles.backArrow}>
        <BackArrow
          testID="Signup.Back"
          navigation={navigation}
          screen="LoginScreen"
          screenName="LoginScreen"
        />
      </View>

      {/* Logo */}
      <View style={styles.logoContainer}>
        <LoginLogo />
      </View>

      {/* Sign up field for email */}
      <View
        style={[styles.inputFieldContainer, { borderWidth: emialInputBorder }]}
      >
        <TextInput
          style={styles.textInput}
          placeholder="UCSD Email"
          keyboardType="default"
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={(text) => {
            setState({
              email: text,
              password: state.password,
              repassword: state.repassword,
            });
          }}
        />
      </View>
      <Text style={styles.errorMsg}>{emailErrorMsg}</Text>

      {/* Sign up field for password */}
      <View
        style={[
          styles.inputFieldContainer,
          { borderWidth: passwordInputBorder },
        ]}
      >
        <TextInput
          style={styles.textInput}
          placeholder="Password"
          keyboardType="default"
          autoCapitalize="none"
          autoCorrect={false}
          secureTextEntry
          onChangeText={(text) => {
            setState({
              email: state.email,
              password: text,
              repassword: state.repassword,
            });
          }}
        />
      </View>
      <Text style={styles.errorMsg}>{passwordErrorMsg}</Text>

      {/* Sign up field for confirm password */}
      <View
        style={[
          styles.inputFieldContainer,
          { borderWidth: repasswordInputBorder },
        ]}
      >
        <TextInput
          style={styles.textInput}
          placeholder="Confirm Password"
          keyboardType="default"
          autoCapitalize="none"
          autoCorrect={false}
          secureTextEntry
          onChangeText={(text) => {
            setState({
              email: state.email,
              password: state.password,
              repassword: text,
            });
          }}
        />
      </View>
      <Text style={styles.errorMsg}>{repasswordErrorMsg}</Text>

      {/* Signin button */}
      <View style={styles.buttonContainer}>
        <SignupButton
          navigation={navigation}
          screenCurr="SignupScreen"
          state={state}
          showMessages={showMessages}
        />
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    width: "100%",
    alignItems: "center",
  },
  backArrow: {
    position: "absolute",
    top: "8%",
    left: "6%",
  },
  logoContainer: {
    marginTop: "30%",
    marginBottom: "10%",
  },
  inputFieldContainer: {
    marginTop: "3%",
    width: "75%",
    height: 42,
    borderRadius: 21,
    borderColor: "red",
    backgroundColor: "white",
    alignItems: "center",
  },
  textInput: {
    height: "100%",
    width: "86%",
  },
  errorMsg: {
    fontSize: 12,
    marginTop: "1.5%",
    color: "red",
    marginLeft: 0,
    left: 0,
    width: "70%",
  },
  buttonContainer: {
    marginTop: "10%",
    width: "75%",
  },
});

export default SignupScreen;
