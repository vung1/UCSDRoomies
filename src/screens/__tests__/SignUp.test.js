import React from "react";

import {
  render,
  fireEvent,
  screen,
  act,
  waitFor,
  mount,
} from "@testing-library/react-native";

import SignupScreen from "../SignupScreen";
import LoginLogo from "../../components/LoginLogo";
import * as renderer from "react-test-renderer";

describe('Rendering Tests', () => {
  it("renders default elements", async () => {
    await render(<SignupScreen />);

    await waitFor(() => {
      expect(screen.getAllByText("SIGN UP WITH UCSD EMAIL").length).toBe(1);
      screen.getByPlaceholderText("UCSD Email");
      screen.getByPlaceholderText("Password");
      screen.getByPlaceholderText("Confirm Password");
    });
  });

  test("matches snapshot", () => {
    const tree = renderer.create(<SignupScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Navigation Tests', () => {
  it("should go back to LoginScreen", async () => {
    const navigation = { navigate: () => {} };
    spyOn(navigation, "navigate");
    // render your component
    const page = render(<SignupScreen navigation={navigation} />);
    // access your button
    const button = page.getByTestId("BackButton");
    // simulate button click
    fireEvent.press(button);
    // expect result
    expect(navigation.navigate).toHaveBeenCalledWith("LoginScreen", {
      screen: "LoginScreen",
    });
  });
});