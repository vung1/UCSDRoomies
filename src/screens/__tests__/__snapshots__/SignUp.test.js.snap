// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Tests matches snapshot 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4285837014,
        4280581583,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "left": "6%",
        "position": "absolute",
        "top": "8%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 50,
          "opacity": 1,
          "width": 50,
        }
      }
      testID="BackButton"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="70%"
        bbWidth="70%"
        focusable={false}
        height="70%"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": "70%",
              "width": "70%",
            },
          ]
        }
        vbHeight={512}
        vbWidth={448}
        width="70%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill={
              Object {
                "payload": 4278190080,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              Array [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": "10%",
        "marginTop": "30%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "red",
          "borderRadius": 21,
          "height": 42,
          "marginTop": "3%",
          "width": "75%",
        },
        Object {
          "borderWidth": 0,
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="UCSD Email"
      style={
        Object {
          "height": "100%",
          "width": "86%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "left": 0,
        "marginLeft": 0,
        "marginTop": "1.5%",
        "width": "70%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "red",
          "borderRadius": 21,
          "height": 42,
          "marginTop": "3%",
          "width": "75%",
        },
        Object {
          "borderWidth": 0,
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "height": "100%",
          "width": "86%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "left": 0,
        "marginLeft": 0,
        "marginTop": "1.5%",
        "width": "70%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "red",
          "borderRadius": 21,
          "height": 42,
          "marginTop": "3%",
          "width": "75%",
        },
        Object {
          "borderWidth": 0,
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Confirm Password"
      secureTextEntry={true}
      style={
        Object {
          "height": "100%",
          "width": "86%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "left": 0,
        "marginLeft": 0,
        "marginTop": "1.5%",
        "width": "70%",
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": "10%",
        "width": "75%",
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "red",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="Signup.Button"
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Arial",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 10,
            }
          }
        >
           SIGN UP WITH UCSD EMAIL 
        </Text>
      </View>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
